integration/tests/epochs/cohort2/epoch_recover_from_efm_test.go

```diff
@@ -0,0 +1,195 @@
+	s.NumOfCollectionClusters = 1
+
+	// run the generic setup, which starts up the network
+	s.BaseSuite.SetupTest()
+}
+
+// getNodeInfoDirs returns the internal node private info dir and the node config dir from a container with the specified role.
+func (s *RecoverEpochSuite) getNodeInfoDirs(role flow.Role) (string, string) {
+	bootstrapPath := s.GetContainersByRole(role)[0].BootstrapPath()
+	internalNodePrivInfoDir := fmt.Sprintf("%s/%s", bootstrapPath, bootstrap.DirPrivateRoot)
+	nodeConfigJson := fmt.Sprintf("%s/%s", bootstrapPath, bootstrap.PathNodeInfosPub)
+	return internalNodePrivInfoDir, nodeConfigJson
+}
+
+// executeEFMRecoverTXArgsCMD executes the efm-recover-tx-args CLI command to generate EpochRecover transaction arguments.
+// Args:
+//
+//	collectionClusters: the number of collector clusters.
+//	numViewsInEpoch: the number of views in the recovery epoch.
+//	numViewsInStakingAuction: the number of views in the staking auction of the recovery epoch.
+//	epochCounter: the epoch counter.
+//	targetDuration: the target duration for the recover epoch.
+//	targetEndTime: the target end time for the recover epoch.
+//
+// Returns:
+//
+//	[]cadence.Value: the transaction arguments.
+func (s *RecoverEpochSuite) executeEFMRecoverTXArgsCMD(
+	collectionClusters int,
+	numViewsInEpoch,
+	numViewsInStakingAuction,
+	recoveryEpochCounter,
+	targetDuration uint64,
+	unsafeAllowOverWrite bool) []cadence.Value {
+	// read internal node info from one of the consensus nodes
+	internalNodePrivInfoDir, nodeConfigJson := s.getNodeInfoDirs(flow.RoleConsensus)
+	snapshot := s.GetLatestProtocolSnapshot(s.Ctx)
+	txArgs, err := run.GenerateRecoverEpochTxArgs(
+		s.Log,
+		internalNodePrivInfoDir,
+		nodeConfigJson,
+		collectionClusters,
+		recoveryEpochCounter,
+		flow.Localnet,
+		numViewsInStakingAuction,
+		numViewsInEpoch,
+		targetDuration,
+		unsafeAllowOverWrite,
+		snapshot,
+	)
+	require.NoError(s.T(), err)
+	return txArgs
+}
+
+// recoverEpoch submits the recover epoch transaction to the network.
+func (s *RecoverEpochSuite) recoverEpoch(env templates.Environment, args []cadence.Value) *sdk.TransactionResult {
+	latestBlockID, err := s.Client.GetLatestBlockID(s.Ctx)
+	require.NoError(s.T(), err)
+
+	tx, err := utils.MakeRecoverEpochTx(
+		env,
+		s.Client.Account(),
+		0,
+		sdk.Identifier(latestBlockID),
+		args,
+	)
+	require.NoError(s.T(), err)
+
+	err = s.Client.SignAndSendTransaction(s.Ctx, tx)
+	require.NoError(s.T(), err)
+	result, err := s.Client.WaitForSealed(s.Ctx, tx.ID())
+	require.NoError(s.T(), err)
+	s.Client.Account().Keys[0].SequenceNumber++
+	require.NoError(s.T(), result.Error)
+
+	return result
+}
+
+// TestRecoverEpoch ensures that the recover epoch governance transaction flow works as expected and a network that
+// enters Epoch Fallback Mode can successfully recover. This test will do the following:
+// 1. Triggers EFM by turning off the sole collection node before the end of the DKG forcing the DKG to fail.
```

Comment:
We have this comment and above we have:
> 	// to manually trigger EFM we assign very short dkg phase len ensuring the dkg will fail

I remember Khalil was trying a few different approaches to reliably trigger EFM, but I'm not sure which one ended up working. We should figure out which of these (Collection Node pause or short DKG phases) is actually causing EFM, and remove the comment and logic for the one that isn't.

Reply:
short DKG phase is somehow flaky, pausing collection node is very reliable though